tests/test_file_download.py

```diff
@@ -186,3 +188,34 @@ def test_hf_hub_download_legacy(self):
         )
         metadata = filename_to_url(filepath, legacy_cache_layout=True)
         self.assertEqual(metadata[1], f'"{DUMMY_MODEL_ID_PINNED_SHA1}"')
+
+    def test_try_to_load_from_cache(self):
+        # Make sure the file is cached
+        filepath = hf_hub_download(DUMMY_MODEL_ID, filename=CONFIG_NAME)
+
+        new_file_path = try_to_load_from_cache(
+            HUGGINGFACE_HUB_CACHE, DUMMY_MODEL_ID, filename=CONFIG_NAME
+        )
+        self.assertEqual(filepath, new_file_path)
+
+        # If file is not cached, returns None
+        self.assertIsNone(
+            try_to_load_from_cache(
+                HUGGINGFACE_HUB_CACHE, DUMMY_MODEL_ID, filename="conf.json"
+            )
+        )
+        # Same for uncached revisions
+        self.assertIsNone(
+            try_to_load_from_cache(
+                HUGGINGFACE_HUB_CACHE,
+                DUMMY_MODEL_ID,
+                filename=CONFIG_NAME,
+                revision="aaa",
+            )
```

Comment:
If possible, I would also had a test case where a legit revision is passed and it works as expected